src/main/java/org/cryptomator/cloudaccess/MetadataRequestAggregator.java

```diff
@@ -15,38 +14,43 @@
 import java.io.InputStream;
 import java.time.Duration;
 import java.time.Instant;
+import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.ExecutionException;
 
-public class MetadataCachingProviderDecorator implements CloudProvider {
+public class MetadataRequestAggregator implements CloudProvider {
 
 	private final static int DEFAULT_CACHE_TIMEOUT_SECONDS = 10;
 
-	final Cache<CloudPath, CompletionStage<CloudItemMetadata>> itemMetadataCache;
+	final Cache<CloudPath, CompletionStage<CloudItemMetadata>> itemMetadataRequestAggregator;
+	final Cache<Map.Entry<CloudPath, Optional<String>>, CompletionStage<CloudItemList>> cloudItemListRequestAggregator;
```

Comment:
I'd rename these caches, as "aggregator" suggest some aggregation logic.

```suggestion
	final Cache<CloudPath, CompletionStage<CloudItemMetadata>> cachedItemMetadataRequests;
	final Cache<Map.Entry<CloudPath, Optional<String>>, CompletionStage<CloudItemList>> cachedItemListRequests;
```